contracts/dual_token/DualStakingToken.sol

```diff
@@ -20,32 +20,34 @@ error ZeroAddress();
 /// @dev Zero value.
 error ZeroValue();
 
-/// @dev Only `owner` has a privilege, but the `sender` was provided.
+/// @dev Only `staker` has a privilege, but the `sender` was provided.
 /// @param sender Sender address.
-/// @param owner Required sender address as an owner.
-error OwnerOnly(address sender, address owner);
+/// @param staker Required staker address.
+error StakerOnly(address sender, address staker);
+
+/// @dev Service is already staked.
+/// @param serviceId Service Id.
+error AlreadyStaked(uint256 serviceId);
+
+/// @dev Wrong service staking state.
+/// @param serviceId Service Id.
+/// @param state Service state.
+error WrongStakingState(uint256 serviceId, IStaking.StakingState state);
 
 /// @dev Caught reentrancy violation.
 error ReentrancyGuard();
 
-struct StakerInfo {
-    // Second token amount
-    uint256 stakingAmount;
-    // Cumulative reward
-    uint256 reward;
-    // Staker account address
-    address account;
-}
-
-/// @title DualStakingToken - Smart contract for dual token staking
+/// @title DualStakingToken - Smart contract for dual token staking: it accepts OLAS-based service NFT
```

Comment:
Do you think we should add a note saying that token is supposed to implement the balanceOf method, or that we assume the token to be an ERC20 ?